src/main/java/org/mockito/internal/debugging/LocationFactory.java

```diff
@@ -24,26 +24,13 @@ private interface Factory {
     }
 
     private static Factory createLocationFactory() {
```

Comment:
I know you were initially working on this PR to get these changes in. However, I would like to include these changes in Mockito 5.1.0, so to avoid pulling in too many changes into 1 go. Can you split out these changes into a separate PR so that we can land that? We can then also consider cleaning up some more of these Java 8 utility classes such as `JavaEightUtil` (https://github.com/mockito/mockito/blob/e2e628918518b73de7381692f28556ac6caa4710/src/main/java/org/mockito/internal/stubbing/defaultanswers/ReturnsEmptyValues.java#L132)

Reply:
@TimvdLippe sure, I can split that, thanks!

Reply:
@TimvdLippe I reverted the changes, would you be open to have a plugin for `LocationFactory` so the alternative implementation could be supplied? (with revert, we are back to reflection <-> `SecurityManager` issues we've started with). I will open separate issue / pull request, should be a low risk change from implementation perspective, but I am wondering how it is aligned with the project.